src/object/obj_rpc.h

```diff
@@ -63,9 +63,9 @@
 	X(DAOS_OBJ_RPC_PUNCH_AKEYS,					\
 		0, &CQF_obj_punch,					\
 		ds_obj_punch_handler, NULL, "akey_punch")		\
-	X(DAOS_OBJ_RPC_QUERY_KEY,					\
-		0, &CQF_obj_query_key,					\
-		ds_obj_query_key_handler, NULL, "key_query")		\
+	X(DAOS_OBJ_RPC_QUERY_KEY_0,					\
```

Comment:
I thought the whole point of the protocols was to actually have multiple versions so something like 

#define QUERY_KEY1 \
     X(....)
     
 #define QUERY_KEY2
     X(...)
     
 #define OBJ_PROTO_RPC_LIST(query_rpc)
   // replaces the query one with query_rpc so it's in same place in new protocol
  
  Anyway, I thought the intent of this stuff was to define one or the other and this should allow you to do that.  Not sure how difficult that is.

Reply:
You would need to be mindful of when you use the full macro and when you just use the query_key2 macro when defining stuff.

Reply:
yes i was experimenting with this, but your idea should work. i'll push the change shortly